enhancements/cloud-integration/aws/aws-sts-support.md

```diff
@@ -0,0 +1,171 @@
+
+```yaml
+      - name: bound-sa-token
+        projected:
+          defaultMode: 420
+          sources:
+          - serviceAccountToken:
+              audience: openshift
+              expirationSeconds: 3600
+              path: token
+```
+
+openshift-install will be modified to accept user provided service account signing keys and configure the Kubernetes APIServer to use them.
+
+For each `CredentialsRequest` the user will need to manually create the relevant Secret and role
+Users will have to manually provision the IAM roles, policies, and Secrets for each `CredentialsRequest`. This process will also require the name of the `ServiceAccount` that the pod consuming the credentials uses, which is typically created by the operator itself. However, we do not presently have the `ServiceAccount` name modeled in the `CredentialsRequest` API, and would like to defer doing this until we have more real world data and experience before making a permanent API commitment. As such, credentials will be documented as a list policy permissions, service account names, and target secrets, which the user will have to iterate:
+
+  1. Create an IAM Role with no policy attached, tagged with `kubernetes.io/cluster/[infraid]=owned` so it can be cleaned up on deprovision. (type = web identity, identity provider created by the admin manually)
+  1. Attach an in-line policy to the role which matches the permissions in the `CredentialsRequest` or documentation.
+  1. Ensure the trust relationship for the Role points to the cluster’s AWS IAM OIDC ARN with conditions to limit which `ServiceAccount` can assume this role.
+     ```json
+{
+	"Version": "2012-10-17",
+		"Statement": [
+		{
+			"Effect": "Allow",
+			"Principal": {
+				"Federated": "<OIDC_IDENTITY_PROVIDER_ARN>"
+			},
+			"Action": "sts:AssumeRoleWithWebIdentity",
+			"Condition": {
+				"StringEquals": {
+					"<OIDC_PROVIDER_URL_without_https://>:sub": "system:serviceaccount:<CredReq spec.Namespace>:<Operator pod ServiceAccount>"
+				}
+			}
+		}
+		]
+}
+```
+  1. Create a secret containing a valid AWS credentials file
+     ```yaml
+apiVersion: v1
+stringData:
+  credentials: |-
+    [default]
+    role_arn=[ AWS Role ARN created for CredentialsRequest ]
+    web_identity_token_file=/var/run/secrets/openshift/serviceaccount/token
+kind: Secret
+metadata:
+  name: [ CredReq .spec.SecretName ]
+  namespace: [ CredReq .spec.Namespace ]
+type: Opaque
+```
+
+Cluster install workflow becomes:
+
+  1. Create an S3 bucket to publish OIDC keys.
+  1. Generate and publish OIDC keys.
+  1. Create an AWS IAM identity provider.
+  1. Configure their local AWS credentials to use STS.
+  1. `openshift-install create install-config`
+  1. Provide their OIDC keys in a manifest, which will be used to configure the Kube APIServer to be used to sign bound service account keys so they are trusted by AWS.
```

Comment:
the keys, secret, issuer are added after `create manifests` not after `create install-config`.

also missing step is updating the install config to be in `credentialMode: Manual`

Reply:
Hmm what's the flow, we need to edit install config to set the mode, is it install-config, then manifests, then cluster?

Reply:
Looks like the above is correct. I will update to include both steps.